### PR TITLE
prototype implementation of narratives for mobile

### PR DESCRIPTION
**Pull request by jameshadfield**
Current status: very early prototype.

Needs extensive testing and addition of features. Not tested with sidebar (theme) customisations.

### CODE REVIEW

File: src/components/narrative/MobileNarrativeDisplay.js [hunk 1]
```diff
@@ -127,6 +127,19 @@ class MobileNarrativeDisplay extends React.Component {
           </MobileContentContainer>
         </>
       );
+    } else if (this.props.currentInFocusBlockIdx === 0) {
+      return (
+        <>
+          <MobileContentContainer height={contentHeight}>
+            {this.pageNarrativeContent()}
```
Comment (jameshadfield) [inline src/components/narrative/MobileNarrativeDisplay.js]: block 0 is the title page & shouldn't display viz / main-markdown (see sketch file for more notes if needed). It'll probably need special styling too to correctly display the header / authors etc.

File: src/components/narrative/MobileNarrativeDisplay.js [hunk 2]
```diff
@@ -153,13 +153,25 @@ class MobileNarrativeDisplay extends React.Component {
...
     );
   }
 
+  NextButton = (props) => (
+    <MobileBannerBottom height={this.state.bannerHeight} onClick={this.goToNextPage}>
+      {props.children}
+    </MobileBannerBottom>
+  )
+
+  PreviousButton = (props) => (
```
Comment (jameshadfield) [inline src/components/narrative/MobileNarrativeDisplay.js]: This is cool - I didn't know fat-arrow functions would bind `this` when used inside classes.

File: src/components/narrative/MobileNarrativeDisplay.js [hunk 3]
```diff
@@ -89,9 +89,7 @@ class MobileNarrativeDisplay extends React.Component {
     }
     const block = this.props.blocks[this.props.currentInFocusBlockIdx];
     return (
-      <div
-        id={`MobileNarrativeBlock_${this.props.currentInFocusBlockIdx}`}
```
Comment (jameshadfield) [inline src/components/narrative/MobileNarrativeDisplay.js]: This is used for scrolling to a particular block via the URL query parameter. While this doesn't always work (bugs separate to this work) I'd like to preserve these id's here please.

**Comment by jameshadfield** [comment]
Just talked through this with Kairsten:

Priorities (we can release after these are done + testing):
* Rebase onto master to pick up yesterday's narratives bugfixes
* Make reloading of a page with `?n=<x>` go to page `<x>` (this now works on non-mobile view, but mobile uses a different implementation)
* Change the explanation text on p0 to reflect mobile behavior
* Fix broken final page (no progress dot, broken links etc)
* Ensure things look OK using nextstrain.org theme extensions

Secondary tasks:
* Tweak the nextstrain.org theme (e.g. add banner colours etc)
* Test breakpoints to switch into mobile view

File: src/components/narrative/MobileNarrativeDisplay.js (hunk 1 above)
Comment (kairstenfay) [reply src/components/narrative/MobileNarrativeDisplay.js]: Is this still relevant? I don't know that I fully understand what you're saying here. Sorry I missed it earlier.

File: src/components/narrative/MobileNarrativeDisplay.js (hunk 2 above)
Comment (kairstenfay) [reply src/components/narrative/MobileNarrativeDisplay.js]: Me neither!

**Comment by jameshadfield** [comment]
Thanks for this work @kairstenfay. I'm merging & releasing this now as it's a huge improvement over previous versions. Will make issues for the remaining bugs / work to do.

**Pull request merged**